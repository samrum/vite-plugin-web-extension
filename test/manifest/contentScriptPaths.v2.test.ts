import { getResourceDir, runManifestV2Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("contentWithChunkedImport");
const resourceDirRoot = `/${resourceDir}`;
const resourceDirRelative = `./${resourceDir}`;

runManifestV2Test("contentScriptPaths", () => ({
  content_scripts: [
    {
      js: [`${resourceDirRoot}/content1.js`],
      matches: ["https://*/*", "http://*/*"],
    },
    {
      js: [`${resourceDirRelative}/content2.js`],
      matches: ["https://*/*", "http://*/*"],
    },
  ],
}));
