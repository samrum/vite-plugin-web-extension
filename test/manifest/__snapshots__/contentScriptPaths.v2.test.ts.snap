// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`contentScriptPaths - Manifest V2 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
export {
  log as l
};
",
    "dynamicImports": [],
    "exports": [
      "l",
    ],
    "facadeModuleId": null,
    "fileName": "assets/log.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": false,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "log",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"content\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/contentWithChunkedImport/content1.js",
    "fileName": "assets/test/manifest/resources/contentWithChunkedImport/content1.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/contentWithChunkedImport/content1.js": {
        "code": "log(\\"content\\");",
        "originalLength": 52,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 15,
      },
    },
    "name": "test/manifest/resources/contentWithChunkedImport/content1",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"content2\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/contentWithChunkedImport/content2.js",
    "fileName": "assets/test/manifest/resources/contentWithChunkedImport/content2.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/contentWithChunkedImport/content2.js": {
        "code": "log(\\"content2\\");",
        "originalLength": 53,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 16,
      },
    },
    "name": "test/manifest/resources/contentWithChunkedImport/content2",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 2,
  \\"content_scripts\\": [
    {
      \\"js\\": [
        \\"test/manifest/resources/contentWithChunkedImport/content1.js\\"
      ],
      \\"matches\\": [
        \\"https://*/*\\",
        \\"http://*/*\\"
      ]
    },
    {
      \\"js\\": [
        \\"test/manifest/resources/contentWithChunkedImport/content2.js\\"
      ],
      \\"matches\\": [
        \\"https://*/*\\",
        \\"http://*/*\\"
      ]
    }
  ],
  \\"web_accessible_resources\\": [
    \\"assets/log.js\\",
    \\"assets/test/manifest/resources/contentWithChunkedImport/content1.js\\",
    \\"assets/test/manifest/resources/contentWithChunkedImport/content2.js\\"
  ]
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/contentWithChunkedImport/content1.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/contentWithChunkedImport/content1.js\\"))})();",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/contentWithChunkedImport/content2.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/contentWithChunkedImport/content2.js\\"))})();",
    "type": "asset",
  },
]
`;
